src/lighteval/utils/cache_management.py

```diff
@@ -377,64 +70,87 @@ def greedy_until(self, docs: List[Doc], ...):
-            task_ids = {cache.get_task_id(doc.task_name, sampling_method) for doc in docs}
-
-            # 1) Identify which samples must be processed because they are not cached
-            docs_not_cached: List[Doc]
-            tasks_with_cached_samples: Set[TaskID]
-            docs_not_cached, tasks_with_cached_samples = cache.get_samples_to_process_and_cache(docs, sampling_method)
-
-            # Log cache statistics
-            cached_count = len(docs) - len(docs_not_cached)
-            if cached_count > 0:
-                logger.info(
-                    f"Cache: {cached_count}/{len(docs)} samples are cached for tasks {', '.join(t_id.task_name for t_id in tasks_with_cached_samples)}"
-                )
-
-            # 2) Process not cached docs and save to file
-            new_results = []
-            if docs_not_cached:
-                tasks_needing_sample_processing = {
-                    cache.get_task_id(doc.task_name, sampling_method) for doc in docs_not_cached
-                }
-                logger.info(
-                    f"Cache: Starting to process {len(docs_not_cached)}/{len(docs)} samples (not found in cache) for tasks {','.join(str(t) for t in tasks_needing_sample_processing)}"
-                )
-                new_results = func(self, docs_not_cached, *args, **kwargs)
-
-                # Store new results in file cache
-                cache.cache_samples(
-                    docs=docs_not_cached,
-                    results=new_results,
-                    task_ids=task_ids,
-                    sampling_method=sampling_method,
-                )
-
-            # 3) Create final results by pulling from newly saved file cache
-            final_cached_results = cache.get_samples_from_cache(docs, task_ids, sampling_method)
-
-            # 4) We only keep samples with the correct sampling method
-            final_results = [
-                s for s in final_cached_results if cache.get_sampling_method(cache._dump_sample(s)) == sampling_method
-            ]
-
-            if any(r is None for r in final_results):
-                raise ValueError("Problem while loading and aggregating items from cache.")
-
-            return final_results
-
-        return wrapper
+    def decorator(sampler: Callable):  # noqa: C901
+        @functools.wraps(sampler)
+        def sync_wrapper(self, docs: Doc | List[Doc], *args, **kwargs):
+            if isinstance(docs, Doc):
+                docs = [docs]
+
+            results = [None] * len(docs)
+            with diskcache.Cache(self.config.cache_dir) as cache:
+                uncached_docs = []
+                uncached_idxs = []
+
+                for idx, doc in enumerate(docs):
+                    key = hash_request(
+                        doc, sampling_method=sampling_method, config=self.config, args=args, kwargs=kwargs
+                    )
+                    if key in cache:
+                        logger.info("Cache hit")
```

Comment:
You're going to get too many logs with this, especially on datasets like mmlu with 10K+ samples - maybe just store how many times the cache was hit and log it at the end? 
`Cache was hit for x documents out of y.`